raiden/network/transport/matrix/transport.py

```diff
@@ -885,8 +911,11 @@ def _get_retrier(self, receiver: Address) -> _RetryQueue:
         if receiver not in self._address_to_retrier:
             retrier = _RetryQueue(transport=self, receiver=receiver)
             self._address_to_retrier[receiver] = retrier
-            if not self._stop_event.ready():
-                retrier.start()
+            # Always start the _RetryQueue, otherwise `stop` will block forever
+            # waiting for the corresponding gevent.Greenlet to complete. This
+            # has no negative side-effects if the transport has stopped because
+            # the retrier itself checks the transport running state.
+            retrier.start()
```

Comment:
I have some concerns about this. This `_get_retrier` is intended to also be used before transport's start, e.g. on `send_async` of `RaidenService._initialize_messages_queues`, then queueing messages but only acting on then once the transport is actually started. As you said, it shouldn't break here, as it'll bail out on first scheduling if transport still isn't running, but I think correct solution would be to not start it here and filter the `wait` list on `stop` for non-started greenlets/runnables.